wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -262,6 +266,31 @@ abstract class Wallet
     sendToOutputs(destinations, feeRate, fromAccount, reserveUtxos)
   }
 
+  override def makeOpReturnCommitment(
+      message: String,
+      hashMessage: Boolean,
+      feeRate: FeeUnit,
+      fromAccount: AccountDb): Future[Transaction] = {
+    val messageToUse = if (hashMessage) {
+      CryptoUtil.sha256(ByteVector(message.getBytes)).bytes
+    } else {
+      if (message.length > 32) {
+        throw new IllegalArgumentException(
+          s"Message cannot be greater than 32 characters, it should be hashed, got $message")
```

Comment:
Are you not allowed to push larger messages?

Reply:
I believe 40 bytes is the max, figured 32 bytes is fine for this so we have a clear distinction between hashes.

If you want to push bigger messages checkout BSV :P

Reply:
Actually looks like it's 83 bytes

https://github.com/bitcoin/bitcoin/blob/master/src/script/standard.h#L33

Reply:
I agree then, this should be changed